frontend/packages/db-structure/src/parser/schemarb/parser.ts

```diff
@@ -26,17 +27,34 @@ import type {
   Tables,
 } from '../../schema/index.js'
 import { aColumn, aRelationship, aTable, anIndex } from '../../schema/index.js'
-import type { Processor } from '../types.js'
+import { type ProcessError, WarningError } from '../errors.js'
+import type { ProcessResult, Processor } from '../types.js'
 import { handleOneToOneRelationships } from '../utils/index.js'
 import { convertColumnType } from './convertColumnType.js'
 
-function extractTableName(argNodes: Node[]): string {
+export class UnsupportedTokenError extends WarningError {
+  constructor(message: string) {
+    super(message)
+    this.name = 'UnsupportedTokenError'
+  }
+}
+
+function extractTableName(
+  argNodes: Node[],
+): Result<string, UnsupportedTokenError> {
   const nameNode = argNodes.find((node) => node instanceof StringNode)
   if (!nameNode) {
-    throw new Error('Table name not found')
+    return err(
+      new UnsupportedTokenError(
+        'Expected a string for the table name, but received different data',
```

Comment:
Table name could be symbol (and we need to support it)

```suggestion
        'Expected string or symbol for the table name, but received different data',
```

Reply:
I see, but it just seems that the timing of the message change would be better to support.

Reply:
Absolutely!